10_datastructures_lab/01.py

```diff
@@ -1,10 +1,15 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+""" Return IPs from hosts file per input machine name(s) """
+import sys
+from collections import defaultdict
+# Read input args
+host_file_path, pc_names = sys.argv[1], sorted(sys.argv[2:])
+# Read hosts into a dictionary
+dict_hosts = defaultdict(str)

```

Comment:
Cool! And the nice thing with defaultdict is you can tell it to return any default value (not just the empty string) using lambda notation. With this one: 
```
dict_hosts = defaultdict(lambda: "No IP")
```

You can simplify the print loop:

```
for pc_name in pc_names: print "%-*s: %s" % (max_length, pc_name, dict_hosts[pc_name])
```